src/encoded/static/components/navigation/components/LeftNav.js

```diff
@@ -166,3 +171,181 @@ const DataNavItemBody = React.memo(function DataNavItemBody(props) {
+        </React.Fragment>
+    );
+
+    return ( // `navItemProps` contains: href, windowHeight, windowWidth, isFullscreen, testWarning, mounted, overlaysContainer
+        <BigDropdownNavItem {...navItemProps} id="search-menu-item" navItemHref="/search" navItemContent={navLink}
+            active={false} autoHideOnClick={false}>
+            <SearchNavItemBody {...bodyProps} />
+        </BigDropdownNavItem>
+    );
+}
+
+const SearchNavItemBody = React.memo(function SearchNavItemBody(props) {
+    const { searchQueryFromHref, searchTypeFromHref } = props;
+
+    const searchTextInputEl = useRef(null);
+    useEffect(() => {
+        if (searchTextInputEl && searchTextInputEl.current) {
+            searchTextInputEl.current.focus();
+        }
+    }, []);
+
+    const initialItemType = AvailableSearchItemTypes[searchTypeFromHref] ? searchTypeFromHref : 'Item';
+    const [searchText, setSearchText] = useState(searchQueryFromHref || '');
+    const [searchItemType, setSearchItemType] = useState(initialItemType);
+    const [searchInputIsValid, setSearchInputIsValid] = useState(true);
+
+    const onChangeSearchItemType = useCallback(function (evtKey) {
+        if (typeof evtKey === 'string') {
+            setSearchItemType(evtKey);
+
+            //validate accession
+            let isValid = true;
+            if (evtKey === 'ByAccession') {
+                isValid = object.isAccessionRegex(searchText);
+            }
+            if (searchInputIsValid !== isValid) {
+                setSearchInputIsValid(isValid);
+            }
+        }
+    });
+    //render hidden form inputs
+    const renderHiddenInputsForURIQuery = useCallback(function () {
+        const query = {};
+        switch (searchItemType) {
+            case 'ExperimentSetReplicate': {
+                const browseBaseParams = navigate.getBrowseBaseParams();
+                _.extend(query, browseBaseParams);
+            }
+                break;
+            case 'ByAccession':
+                break;
+            default: {
+                _.extend(query, { 'type': searchItemType });
+            }
+                break;
+        }
+        return SearchBar.renderHiddenInputsForURIQuery(query);
+    });
+    //custom placeholder text for search item type
+    const getSearchTextPlaceholder = useCallback(function () {
+        switch (searchItemType) {
+            case 'Item':
+                return 'Search 4DN Data Portal';
+            case 'ByAccession':
+                return 'Type Item\'s Accession (e.g. 4DNXXXX ...)';
+            default:
+                return "Search in " + AvailableSearchItemTypes[searchItemType].text;
+        }
+    });
+    //navigate to Item page directly without searching
+    const navigateByAccession = useCallback(function (evt) {
+        if (searchItemType === 'ByAccession') {
+            evt.preventDefault();
+            evt.stopPropagation();
+            navigate('/' + searchText);
+        }
+    });
```

Comment:
`useCallback(function eventHandlerFunction(e){ ... } })` is a shorthand form of `useMemo(function(){ return eventHandlerFunction(e){ ... }; })`. So probably need to have `[ searchItemType ]` as 2nd param for these, otherwise `getSearchTextPlaceholder` might always return same thing even if value of searchItemType has changed. To improve performance could theoretically use a useMemo and return multiple functions from it so that 'searchItemType' doesn't have to get compared for every function, i.e. 

```
const { getSearchTextPlaceholder, navigateByAccession } = useMemo(function(){
    return {
        getSearchTextPlaceholder: function () {
            switch (searchItemType) {
                case 'Item':
                    return 'Search 4DN Data Portal';
                case 'ByAccession':
                    return 'Type Item\'s Accession (e.g. 4DNXXXX ...)';
                default:
                    return "Search in " + AvailableSearchItemTypes[searchItemType].text;
            }
        },
        navigateByAccession: function (evt) {
            if (searchItemType === 'ByAccession') {
                evt.preventDefault();
                evt.stopPropagation();
                navigate('/' + searchText);
            }
        },
        ........
    };
}, [ searchItemType ]);
```